log/src/main/java/kcl/teamIndexZero/traffic/log/FileIO/FileIO.java

```diff
@@ -52,69 +59,90 @@ public boolean createFile() {
     /**
      * Deletes the file if it exists
      * return Success
-     * @exception InvalidPathException when the path description is invalid
-     * @exception IOException when the file cannot be accessed
+     *
+     * @throws InvalidPathException when the path description is invalid
+     * @throws IOException          when the file cannot be accessed
      */
-    public boolean deleteFile() throws InvalidPathException, IOException {
+    public synchronized boolean deleteFile() throws InvalidPathException, IOException {
         try {
-            if( Files.exists( this.getFilePath() )) {
-                Files.delete( this.getFilePath() );
+            if (Files.exists(this.getFilePath())) {
+                Files.delete(this.getFilePath());
```

Comment:
The delete function is only really used in the unit testing and is there also in case it's needed down the road for something else. The logger never ever deletes anything itself. It just creates a new file with the launch time-stamp appended to the name each time.